clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -998,15 +998,18 @@ public static void closeAll(Closeable... closeables) throws IOException {
             throw exception;
     }
 
-    public static void swallow(
-        Logger log,
-        String what,
-        Runnable runnable
-    ) {
-        try {
-            runnable.run();
-        } catch (Throwable e) {
-            log.warn("{} error", what, e);
+    /**
+     * Run the supplied code. If an exception is thrown, it is swallowed and registered to the firstException parameter.
+     */
+    public static void swallow(Logger log, String what, final Runnable code, final AtomicReference<Throwable> firstException) {
+        if (code != null) {
+            try {
+                code.run();
+            } catch (Throwable t) {
+                log.warn("{} error", what, t);
```

Comment:
Should this be an error instead of a warn to be consistent with `closeQuietly`? Moreover, I wonder if we could improve the error message. We would get something like `fetcher close error` which is not really inline with what we usually log. For instance, `closeQuietly` would log something like `Failed to close fetch...`. Do you have any thoughts on this?

Reply:
Fair point. Though not all "swallows" will require error level. So, I created a generic logging function based on `CoreUtils.swallow` and then used error for this specific instance of closing fetcher and coordinator. I have also updated the comment.

Let me know if that look right? Happy to change it further.